### PR TITLE
Optimize process_sample() Function Using Pandas DataFrame

### PR DESCRIPTION
**Pull request by jylee-bcm**
The previous implementation of process_sample()
relied on Python built-in for-loops, which were
less efficient. This update replaces the for-loops with optimized Pandas DataFrame operations
to enhance performance.

#### Before the change
image
#### After the change 
image

### Validated if the outcome is identical
The output is confiormed as it's almost identical.
In detail, nearly 130 rows are different among 10800 rows, but the difference is very subtle and ignorable.
image

[1.improved_df.csv](https://github.com/user-attachments/files/15922289/1.improved_df.csv)
[1.improved_slow.csv](https://github.com/user-attachments/files/15922287/1.improved_slow.csv)

### CODE REVIEW

File: bin/extraModel/generate_bivar_data.py [hunk 1]
```diff
@@ -184,7 +185,132 @@ def generate_feature_vector(i, j):
 
     if len(bivar_feature_mats) > 0:
         bivar_feature_mats = pd.concat(bivar_feature_mats, axis=0)
-        bivar_feature_mats = bivar_feature_mats[~bivar_feature_mats.index.duplicated()]
+        # bivar_feature_mats = bivar_feature_mats[~bivar_feature_mats.index.duplicated()]
```
Comment (hyunhwan-bcm) [inline bin/extraModel/generate_bivar_data.py]: Remove the line unless it will be used.

File: bin/extraModel/generate_bivar_data.py [hunk 2]
```diff
@@ -8,8 +9,7 @@
 import shutil
 
 
-def process_sample(data_folder, sample_id, default_pred, labeling=False):
-
+def process_sample_old(data_folder, sample_id, default_pred, labeling=False):
```
Comment (hyunhwan-bcm) [inline bin/extraModel/generate_bivar_data.py]: Remove it unless it is being used. Leaving this makes it hard for me to review.

**Review by hyunhwan-bcm — CHANGES_REQUESTED** [review]
Please update accordingly to the comments.

File: bin/extraModel/generate_bivar_data.py [hunk 3]
```diff
@@ -46,145 +43,91 @@ def process_sample(data_folder, sample_id, default_pred, labeling=False):
...
-    print(
-        f"Now starting to generate recessive feature matrix for {len(gene_dict)} genes, {feature_df.shape[0]} variants."
+    varId_geneEnsId_df = df[['geneEnsId']]
+    varId_geneEnsId_df.index.name = 'varId'
+
+    geneEnsId_varId_df = varId_geneEnsId_df[varId_geneEnsId_df.geneEnsId.str.startswith("ENSG")] \
+        .reset_index() \
+        .sort_values(['geneEnsId', 'varId']) \
+        .drop_duplicates()
+    geneEnsId_varId_df.groupby('geneEnsId').value_counts()
```
Comment (jylee-bcm) [inline bin/extraModel/generate_bivar_data.py]: I will remove this line soon

**Comment by jylee-bcm** [comment]
image
Pandas believes the result files are identical.

**Pull request merged**